### ISSUE TITLE
twisted.words.test.test_irc.CTCPQueryTests.test_ctcpQuery_CLIENTINFO relies on stable dict ordering

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8015]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8015|
|Type|enhancement|
|Created|2015-09-04 05:53:11Z|
|Branch|https://github.com/twisted/twisted/tree/clientinfo-pypy-8015|

This breaks PyPy, and Twisted tested under tox.

<details><summary>Searchable metadata</summary>

```
trac-id__8015 8015
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__PyPy_support PyPy-support
branch__branches_clientinfo_pypy_8015 branches/clientinfo-pypy-8015
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1441345991388293 1441345991388293
changetime__1441514715313863 1441514715313863
version__None None
owner__hawkowl hawkowl
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [45627]) Branching to clientinfo-pypy-8015.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

Happy with the builds. See https://buildbot.twistedmatrix.com/builders/ubuntu15.04-pypy25/builds/34/steps/select/logs/problems (ran on this branch), it does not have the test failure (compare with trunk, https://buildbot.twistedmatrix.com/builders/ubuntu15.04-pypy25/builds/32/steps/select/logs/problems ).

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Changes look good.
I don't see the news fragment :)

Just a minor comment.

Do we really need to sort the response of CLIENTINFO?
It might be a waste of CPU just to please a test.
I was expecting that this patch will just update the test... for example by spliting the result and using assertItemsEqual


Please review my comments and feel free to merge as you wish ... but I think that the problem is with the test and not the implementation :)

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

There's a misc topfile -- it shows in the GitHub browser.

I asked Glyph about this, and we've deduced that in situations where dict ordering is unstable, sorting is better -- you can tell the Python implementation from the dict ordering, so we should take mild efforts to make that not obvious, for security reasons.

Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45633]) Merge clientinfo-pypy-8015: Make twisted.words.test.test_irc.CTCPQueryTests.test_ctcpQuery_CLIENTINFO not rely on stable dict ordering

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8015](https://github.com/twisted/twisted/issues/8015)
